src/ListBuilder.php

```diff
@@ -0,0 +1,176 @@
+ */
+class ListBuilder implements ListBuilderInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The pager.
+   *
+   * @var \Drupal\Core\Pager\PagerManagerInterface
+   */
+  protected $pager;
+
+  /**
+   * The entity repository.
+   *
+   * @var \Drupal\Core\Entity\EntityRepositoryInterface
+   */
+  protected $entityRepository;
+
+  /**
+   * The form builder.
+   *
+   * @var \Drupal\Core\Form\FormBuilderInterface
+   */
+  protected $formBuilder;
+
+  /**
+   * The list execution manager.
+   *
+   * @var \Drupal\oe_list_pages\ListExecutionManagerInterface
+   */
+  protected $listExecutionManager;
+
+  /**
+   * ListBuilder constructor.
+   *
+   * @param \Drupal\oe_list_pages\ListExecutionManagerInterface $listExecutionManager
+   *   The list execution manager.
+   * @param \Drupal\Core\Entity\EntityTypeManager $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Pager\PagerManagerInterface $pager
+   *   The pager.
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entityRepository
+   *   The entity repository.
+   * @param \Drupal\Core\Form\FormBuilderInterface $formBuilder
+   *   The form builder.
+   */
+  public function __construct(ListExecutionManagerInterface $listExecutionManager, EntityTypeManager $entityTypeManager, PagerManagerInterface $pager, EntityRepositoryInterface $entityRepository, FormBuilderInterface $formBuilder) {
+    $this->listExecutionManager = $listExecutionManager;
+    $this->entityTypeManager = $entityTypeManager;
+    $this->pager = $pager;
+    $this->entityRepository = $entityRepository;
+    $this->formBuilder = $formBuilder;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildList(ContentEntityInterface $entity): array {
+
+    $build = [];
+    $cache = new CacheableMetadata();
+    $cache->addCacheTags($entity->getEntityType()->getListCacheTags());
+
+    $list_execution = $this->listExecutionManager->executeList($entity);
+    if (empty($list_execution)) {
+      $cache->applyTo($build);
+      return $build;
+    }
+
+    $list_source = $list_execution->getListSource();
+    if (!$list_source) {
+      $cache->applyTo($build);
+      return $build;
+    }
+
+    $query = $list_execution->getQuery();
+    $result = $list_execution->getResults();
+    $wrapper = $list_execution->getListPluginWrapper();
+
+    // Determine the view mode to render with and the sorting.
+    $bundle_entity_type = $this->entityTypeManager->getDefinition($wrapper->getSourceEntityType())->getBundleEntityType();
+    $storage = $this->entityTypeManager->getStorage($bundle_entity_type);
+    $bundle = $storage->load($wrapper->getSourceEntityBundle());
+    $view_mode = $bundle->getThirdPartySetting('oe_list_pages', 'default_view_mode', 'teaser');
+    $cache->addCacheableDependency($query);
+    $cache->addCacheableDependency($bundle_entity_type);
```

Comment:
Wrong cache dependency here, you added the definition and not the bundle entity.